icu4c/source/i18n/calendar.cpp

```diff
@@ -392,6 +397,12 @@ static Calendar *createStandardCalendar(ECalType calType, const Locale &loc, UEr
         case CALTYPE_DANGI:
             cal.adoptInsteadAndCheckErrorCode(new DangiCalendar(loc, status), status);
             break;
+        case CALTYPE_TIBETAN:
+            cal.adoptInsteadAndCheckErrorCode(new TibetanCalendar(loc, status, TibetanCalendar::PHUGPA), status);
```

Comment:
In the past we have this strange combination of subclassing + passing a enum then if check way to implement different calendar which behave closely. I recently refactor the other Calendar subclass to remove the C style if check way and now purely use subclassing. Could you change your code so TibetanCalendar is implement only TibetanCalendar::PHUGPA and add a subclass TibetanTsurphCalendar to implement TibetanCalendar::TSURPHU and put the differences in the override function instead. (see https://github.com/unicode-org/icu/pull/2215 )